test/functional/tests/ci/test_ci_write_mode_write_around.py

```diff
@@ -0,0 +1,371 @@
+# SPDX-License-Identifier: BSD-3-Clause
+#
+
+import pytest
+import time
+
+from api.cas import casadm
+from core.test_run import TestRun
+from storage_devices.disk import DiskType, DiskTypeSet, DiskTypeLowerThan
+from test_utils.os_utils import Udev
+from test_utils.size import Unit, Size
+from test_tools.dd import Dd
+from test_tools.iostat import IOstatBasic
+
+
+@pytest.mark.require_disk("cache", DiskTypeSet([DiskType.nand, DiskType.optane]))
+@pytest.mark.require_disk("core", DiskTypeLowerThan("cache"))
+@pytest.mark.CI()
+def test_ci_write_around_read():
+    """
+    title: Verification test for write mode: write around
+    description: Verify if write mode: write around, works as expected and cache only reads
+    and does not cache write
+    pass criteria:
+    - writes are not cached
+    - reads are cached
+    """
+
+    with TestRun.step("Prepare devices"):
+        cache_device = TestRun.disks["cache"]
+        core_device = TestRun.disks["core"]
+
+        cache_device.create_partitions([Size(1, Unit.GibiByte)])
+        core_device.create_partitions([Size(2, Unit.GibiByte)])
+
+        cache_device = cache_device.partitions[0]
+        core_device = core_device.partitions[0]
+
+    with TestRun.step("Start cache in Write Around mode"):
+        count = 100
+
+        cache = casadm.start_cache(cache_dev=cache_device, cache_id=1, force=True,
+                                   cache_mode=casadm.CacheMode.WA)
+        casadm.add_core(cache, core_device)
+        time.sleep(2.5)
+
+    with TestRun.step("Test read from cache"):
+        Udev.disable()
+        dd = (
+            Dd()
+            .input("/dev/cas1-1")
+            .output("/dev/null")
+            .count(count)
+            .block_size(Size(1, Unit.Blocks4096))
+            .iflag("direct")
+        )
+        dd.run()
+
+        iostat = IOstatBasic.get_iostat_list([cache_device.parent_device])
+        read_cache_1 = iostat[0].total_reads
+
+        dd = (
+            Dd()
+            .input("/dev/cas1-1")
+            .output("/dev/null")
+            .count(count)
+            .block_size(Size(1, Unit.Blocks4096))
+            .iflag("direct")
+        )
+        dd.run()
+
+        iostat = IOstatBasic.get_iostat_list([cache_device.parent_device])
+        read_cache_2 = iostat[0].total_reads
+
+        Udev.enable()
+
+    with TestRun.step("Check if reads are cached"):
+        read_cache_delta = read_cache_2 - read_cache_1
+        data_read = Size(count, Unit.Blocks4096)
+        if read_cache_delta == data_read:
+            TestRun.LOGGER.info(f"Blocks read from cache:{read_cache_delta} == {data_read}")
+        else:
+            TestRun.LOGGER.error(f"Blocks read from cache:{read_cache_delta} != {data_read}")
+
+        cache.stop()
+
+
+@pytest.mark.require_disk("cache", DiskTypeSet([DiskType.nand, DiskType.optane]))
+@pytest.mark.require_disk("core", DiskTypeLowerThan("cache"))
+@pytest.mark.CI()
+def test_ci_write_through_read():
```

Comment:
To be honest I think you don't need to add two separate tests for WT and WA. It can be a single test with a properly adapted verification code. You can use trait cache modes API to recognize which cache modes are supposed to insert data on reads and which are not

Reply:
They are two different cases which is why they are separate tests. They also used to be one before the rewrite, which made it overcomplicated and difficult to grasp - separating them was a decision made to improve readability among other things.

Reply:
It's one test which is supposed to be parametrized by cache modes, actually. Apart from different cache modes, `test_ci_write_through_read` and `test_ci_write_around_read` are exactly the same. I would say leaving them as two separate tests is counterintuitive as a reviewer will be expecting two different scenarios in that case but there is only one. Refactoring them into one test will save time to the reviewers and people investigating the test it if the test fails in the future.